packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraStateFlutterApiWrapper.java

```diff
@@ -0,0 +1,135 @@
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+  private CameraStateFlutterApi cameraStateFlutterApi;
+
+  /**
+   * Constructs a {@link CameraStateFlutterApiWrapper}.
+   *
+   * @param binaryMessenger used to communicate with Dart over asynchronous messages
+   * @param instanceManager maintains instances stored to communicate with attached Dart objects
+   */
+  public CameraStateFlutterApiWrapper(
+      @NonNull BinaryMessenger binaryMessenger, @NonNull InstanceManager instanceManager) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+    cameraStateFlutterApi = new CameraStateFlutterApi(binaryMessenger);
+  }
+
+  /**
+   * Stores the {@link CameraState} instance and notifies Dart to create and store a new {@link
+   * CameraState} instance that is attached to this one. If {@code instance} has already been added,
+   * this method does nothing.
+   */
+  public void create(
+      @NonNull CameraState instance,
+      @NonNull CameraState.Type type,
+      @Nullable CameraState.StateError error,
+      @NonNull CameraStateFlutterApi.Reply<Void> callback) {
+
+    // Convert CameraX CameraState.Type to CameraStateType that the Dart side understands.
+    CameraStateType cameraStateType = CameraStateType.CLOSED;
+    switch (type) {
+      case CLOSED:
+        cameraStateType = CameraStateType.CLOSED;
+        break;
+      case CLOSING:
+        cameraStateType = CameraStateType.CLOSING;
+        break;
+      case OPEN:
+        cameraStateType = CameraStateType.OPEN;
+        break;
+      case OPENING:
+        cameraStateType = CameraStateType.OPENING;
+        break;
+      case PENDING_OPEN:
+        cameraStateType = CameraStateType.PENDING_OPEN;
+        break;
+    }
+
+    if (error != null) {
+      // We need to create a CameraStateError if there is a problem with the current camera
+      // state to send to the Dart side.
+      new CameraStateErrorFlutterApiWrapper(binaryMessenger, instanceManager)
+          .create(
+              error,
+              Long.valueOf(error.getCode()),
+              getCameraStateErrorDescription(error),
+              reply -> {});
+    }
+
+    if (!instanceManager.containsInstance(instance)) {
+      cameraStateFlutterApi.create(
+          instanceManager.addHostCreatedInstance(instance),
+          new CameraStateTypeData.Builder().setValue(cameraStateType).build(),
+          instanceManager.getIdentifierForStrongReference(error),
+          callback);
+    }
+  }
+
+  /**
+   * Returns an error message corresponding to the specified {@link CameraState.StateError}.
+   *
+   * <p>See https://developer.android.com/reference/androidx/camera/core/CameraState#constants_1 for
+   * more information on the different {@link CameraState.StateError} types.
+   */
+  private String getCameraStateErrorDescription(@NonNull CameraState.StateError cameraStateError) {
```

Comment:
Can you include who this error string is for? Ex for developers, can it be shown to users, can it be used for error handling etc? 

My guess is it is a Developer visible string and should not be used in error handling.